src/Features/Core/Portable/ConvertForEachToFor/AbstractConvertForEachToForCodeRefactoringProvider.cs

```diff
@@ -150,29 +155,12 @@ public override async Task ComputeRefactoringsAsync(CodeRefactoringContext conte
                 return null;
             }
 
-            // VB can have Next variable. but we only support
-            // simple 1 variable case.
-            if (operation.NextVariables.Length > 1)
+            if (!CheckIfNextStatementIsSupported(foreachVariable, operation, foreachStatement))
```

Comment:
Can you please add comments here and on the definition of `CheckIfNextStatementIsSupported` what the method is supposed to do? It is not very clear from just the name of the method.